src/auth/negotiate/kerberos/negotiate_kerberos_pac.cc

```diff
@@ -362,6 +367,114 @@ getextrasids(char *ad_groups, uint32_t ExtraSids, uint32_t SidCount)
     return ad_groups;
 }
 
+static char *
+get_resource_group_domain_sid(uint32_t ResourceGroupDomainSid) {
+
+    if (ResourceGroupDomainSid!= 0) {
+        uint8_t rev;
+        uint64_t idauth;
+        char dli[256];
+        char *ag;
+        int l;
+
+        align(4);
+
+        uint32_t nauth = get4byt();
+
+        size_t length = 1+1+6+nauth*4;
+
+        ag=(char *)xcalloc((length+1)*sizeof(char),1);
+        // the first byte is a length of the SID
+        ag[0] = (char) length;
```

Comment:
The binary value of length is 4 bytes long. Please validate that the stored value will fit into this 1-byte space without truncation.

Reply:
I changed the store space to 4-bytes in a new file version.